extend docs (#69)
openvinotoolkit/openvino.genai: README.md

## GenAI Pipeline Repository

The GenAI repository contains pipelines that implement image and text generation tasks.
The implementation uses OpenVINO capabilities to optimize the pipelines. Each sample covers
a family of models and suggests certain modifications to adapt the code to specific needs.
It includes the following pipelines:

1. [Benchmarking script for large language models](./llm_bench/python/)
2. [Casual LM](./text_generation/casual_lm/cpp/)
3. [OpenVINO Stable Diffuison (with LoRA) C++ pipeline](./image_generation/stable_diffusion_1_5/cpp/)

> [!NOTE]
> This project is not for production use.

### License

The GenAI repository is licensed under [Apache License Version 2.0](LICENSE).
By contributing to the project, you agree to the license and copyright terms therein and release
your contribution under these terms.

## Requirements

Requirements may vary for different samples. See respective readme files for more details,
and make sure to install the OpenVINO version listed there. Refer to documentation to see
[how to install OpenVINO](docs.openvino.ai/install).

> To enable non ASCII characters for Windows cmd open `Region` settings from `Control panel`.
> `Administrative`->`Change system locale`->`Beta: Use Unicode UTF-8 for worldwide language support`->`OK`.
> Reboot for the change to take effect.

Author: ```suggestion

```

## GenAI Pipeline Repository

The GenAI repository contains pipelines that implement image and text generation tasks.
The implementation uses OpenVINO capabilities to optimize the pipelines. Each sample covers
a family of models and suggests certain modifications to adapt the code to specific needs.
It includes the following pipelines:

1. [Benchmarking script for large language models](./llm_bench/python/)
2. [Casual LM](./text_generation/casual_lm/cpp/)
3. [OpenVINO Stable Diffuison (with LoRA) C++ pipeline](./image_generation/stable_diffusion_1_5/cpp/)

> [!NOTE]
> This project is not for production use.

### License

The GenAI repository is licensed under [Apache License Version 2.0](LICENSE).
By contributing to the project, you agree to the license and copyright terms therein and release
your contribution under these terms.

## Requirements

Requirements may vary for different samples. See respective readme files for more details,
and make sure to install the OpenVINO version listed there. Refer to documentation to see
[how to install OpenVINO](docs.openvino.ai/install).

> supporting C++ or Python APIs.
> 
> To build the pipelines, including the `user_ov_extensions` submodule, you can use commands
> like these:
> 
> ```sh
> git submodule update --init
> mkdir ./build/ && cd ./build/
> source <OpenVINO dir>/setupvars.sh
> cmake -DCMAKE_BUILD_TYPE=Release ../ && cmake --build ./ --config Release -j
> ```

Author: ```suggestion

```

<!---
- [Repository Description](#repository-description))
   - [Pipeline Samples](#pipeline-samples)
   - [License](#license)
- [Requirements](#requirements)

> -->

Author: This is just an editing leftover. Depending on the length of the article, we may consider adding a table of contents at the top. With this few sections, it would not make sense. We can remove the comment and bring it back if the readme gets longer.

> -->

Reviewer: Ok. Please remove it for now.

> <!---
> - [Repository Description](#repository-description))
>    - [Pipeline Samples](#pipeline-samples)
>    - [License](#license)
> - [Requirements](#requirements)
> -->

Author: ```suggestion

-->
```

## GenAI Pipeline Repository

The GenAI repository contains pipelines that implement image and text generation tasks.
The implementation uses OpenVINO capabilities to optimize the pipelines. Each sample covers
a family of models and suggests certain modifications to adapt the code to specific needs.
It includes the following pipelines:

1. [Benchmarking script for large language models](./llm_bench/python/)
2. [Casual LM](./text_generation/casual_lm/cpp/)
3. [OpenVINO Stable Diffuison (with LoRA) C++ pipeline](./image_generation/stable_diffusion_1_5/cpp/)

> [!NOTE]
> This project is not for production use.

### License

The GenAI repository is licensed under [Apache License Version 2.0](LICENSE).
By contributing to the project, you agree to the license and copyright terms therein and release
your contribution under these terms.

## Requirements

Requirements may vary for different samples. See respective readme files for more details.

> To use the pipelines install a [proper OpenVINO version](docs.openvino.ai/install), 

Reviewer: Every pipeline describes which version they require. Why would you introduce such an ambiguous instruction in root readme in first place? The same goes to `To build the pipelines, including the user_ov_extensions submodule, you can use commands like these:`

Author: As mentioned in the previous comment, it is to indicate that people need to check the readme file for the chosen sample to get the requirements. If there are any requirements common to all samples, they should be mentioned here. If there are no such requirements, we can just state "go and check readme files for each sample you want to use."
The info on the user_ov_extensions comes from the original content I used as the base. If you think it is irrelevant, let's remove it.

Reviewer: Building user_ov_extensions is required only for C++ pipelines. My proposal is to keep "an OpenVINO version referred in the sample" general message, but remove build instruction from here.

> To use the pipelines install a [proper OpenVINO version](docs.openvino.ai/install), 

Author: It could say "an OpenVINO version supporting the API used in the sample"
This is to remind users that OpenVINO comes in different flavors.

> To use the pipelines install a [proper OpenVINO version](docs.openvino.ai/install), 

Reviewer: It's still impossible to tell which OpenVINO version supports the required API.

Maybe "an OpenVINO version referred in the sample"

supporting C++ or Python APIs.

To build the pipelines, including the `user_ov_extensions` submodule, you can use commands
like these:

```sh
git submodule update --init
mkdir ./build/ && cd ./build/
source <OpenVINO dir>/setupvars.sh
cmake -DCMAKE_BUILD_TYPE=Release ../ && cmake --build ./ --config Release -j
```

> To enable non ASCII characters for Windows cmd open `Region` settings from `Control panel`.

Reviewer: ```suggestion
To enable Unicode characters for Windows cmd open `Region` settings from `Control panel`.
```

It was changed in https://github.com/openvinotoolkit/openvino.genai/pull/9. If you are going to duplicate this

Author: Oh, so it's been removed. Nah, if you consider it unnecessary, let's remove it. It was present in the original readme version I started editing for the PR.

Reviewer: As far as I know Python doesn't have such issues, so `llm_bench` doesn't require that. That's the reason I moved it to a specific pipeline. I prefer to remove it from root readme

<!---
- [Repository Description](#repository-description))
   - [Pipeline Samples](#pipeline-samples)
   - [License](#license)
- [Requirements](#requirements)
-->
## GenAI Pipeline Repository

The GenAI repository contains pipelines that implement image and text generation tasks.
The implementation uses OpenVINO capabilities to optimize the pipelines. Each sample covers
a family of models and suggests certain modifications to adapt the code to specific needs.

> Note
This project is not for production use.

### Pipeline Samples 
 - [image generation by stable diffusion](image_generation/stable_diffusion_1_5/cpp)
 - [large language model demo](llm/cpp)
 - [Benchmarking script for large language models](llm_bench/python)

### License

The GenAI repository is licensed under [Apache License Version 2.0](LICENSE).
By contributing to the project, you agree to the license and copyright terms therein and release
your contribution under these terms.

## Requirements

Requirements may vary for different samples. See respective readme files for more details.

> Every pipeline requires https://github.com/openvinotoolkit/openvino for C++ to be installed.

Reviewer 2: For the python sample, C++ OpenVINO is not required.

Author: Yeah, I was wondering if this statement was still true :) Thanks.
